server/lib/s3backup/__init__.py

```diff
@@ -0,0 +1,116 @@
+"""
+This module provides convenience functions that interface to lower-level services, provided by the boto3 module.
+"""
+import boto3
+import os
+import sys
+import glob
+import base64
+import hashlib
+import shutil
+import time
+import configtools
+from datetime import datetime
+from configparser import ConfigParser
+
+class Entry(object):
+    """
+    An Entry object consists of a name (of an S3 object) and the MD5 value of that object. It is used to create an object with the name and MD5 value, so that it can be compared with other similar objects.
+    """
+    def __init__(self, name, md5):
+        self.name = name
+        self.md5 = md5
+
+    def __eq__(self, other):
+        return self.name == other.name and self.md5 == other.md5
+
+
+class S3Connector(object):
+    def __init__(self, access_key_id, secret_access_key, endpoint_url, bucket_name):
+        self.s3connector = boto3.client('s3',
+                                        aws_access_key_id='{}'.format(
+                                            access_key_id),
+                                        aws_secret_access_key='{}'.format(
+                                            secret_access_key),
+                                        endpoint_url='{}'.format(endpoint_url))
+
+    def ts(self, *args):
+        return time.strftime("%Y-%m-%dT%H:%M:%S-%Z", time.localtime())
+
+    def list_buckets(self):
+        try:
+            list_bucket = self.s3connector.list_buckets()
+            buckets = [bucket['Name'] for bucket in list_bucket['Buckets']]
+            return buckets
+        except Exception:
```

Comment:
This is a pretty generic exception.  Can you see if there is a base exception class the boto3 uses so that we can catch to differentiate between boto3 client errors and other more generic python errors?

For example, if `bucket` does not have a `Name` entry we'll get a `KeyError`, same for `list_bucket` not having a `Buckets` entry.

You could consider doing the following to isolate the errors where they are raised:

```
try:
    bucket_list = self.s3connector.list_buckets()
except Exception:
    buckets = []
else:
    try:
        buckets = [bucket['Name'] for bucket in bucket_list['Buckets']]
    except Exception:
        logger.exception("Unable to process list of buckets from S3")
return buckets
```

Reply:
Dropping the function altogether, since we changed the upload implementation which was using the function `list_buckets`